src/class-parsely.php

```diff
@@ -694,8 +694,14 @@ private function get_categories( int $post_id, string $delimiter = '/' ): array
 		if ( $last_tag ) {
 			$tags = explode( '/', $last_tag );
 		}
-		// remove uncategorized value from tags.
-		return array_diff( $tags, array( 'Uncategorized' ) );
+
+		// Remove default category name from tags if needed.
+		$default_category_name = get_cat_name( get_option( 'default_category' ) );
+		if ( null !== $default_category_name ) {
```

Comment:
We should check for the type of the option returned in `get_option`. `get_cat_name` only accepts integers. Then, we don't need to check if that's null, because it will never be. `get_cat_name` always returns a string. If it can't find the category, it will return an empty string.

Reply:
How about something like:
```
$default_category_name = get_cat_name( (int) get_option( 'default_category' ) );
```
or:
```
$default_category_name = get_cat_name( intval( get_option( 'default_category' ) ) );
```
I prefer the first one.

Reply:
WordPress core generally doesn't add fallback values or casting for when `get_option( 'default_category' );` returns anything but an integer - it's not something that can be unset, short of running WP-CLI command or editing the database directly.

There is just [one place](https://github.com/WordPress/WordPress/blob/56d10309d0a20b877d8f50b48b5e02988e3954aa/wp-includes/taxonomy.php#L1974) where it is typecast, and that is for protecting against it being deleted.

While I'm in favour of adding useful protections and guards, it doesn't seem relevant here. If the default_option needs deleting, then we want it to shout and scream and be a fatal if necessary, as that's a bigger issue that needs to be fixed; covering it up doesn't help.